api/src/main/java/org/openmrs/module/ugandaemrreports/definition/cohort/evaluator/LongRefillsCohortEvaluator.java

```diff
@@ -0,0 +1,59 @@
+package org.openmrs.module.ugandaemrreports.definition.cohort.evaluator;
+
+import org.joda.time.LocalDate;
+import org.openmrs.Obs;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.reporting.cohort.EvaluatedCohort;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.evaluator.CohortDefinitionEvaluator;
+import org.openmrs.module.reporting.common.DateUtil;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.module.reporting.evaluation.querybuilder.HqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.openmrs.module.ugandaemrreports.common.StubDate;
+import org.openmrs.module.ugandaemrreports.definition.cohort.definition.ArtStartCohortDefinition;
+import org.openmrs.module.ugandaemrreports.definition.cohort.definition.LongRefillsCohortDefinition;
+import org.openmrs.module.ugandaemrreports.metadata.HIVMetadata;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Date;
+import java.util.List;
+
+import static org.openmrs.module.ugandaemrreports.common.Helper.getDates;
+
+/**
+ */
+@Handler(supports = {LongRefillsCohortDefinition.class})
+public class LongRefillsCohortEvaluator implements CohortDefinitionEvaluator {
+
+    @Autowired
+    EvaluationService evaluationService;
+
+    @Autowired
+    HIVMetadata hivMetadata;
+
+    @Override
+    public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext context)
+            throws EvaluationException {
+        EvaluatedCohort ret = new EvaluatedCohort(cohortDefinition, context);
+        LongRefillsCohortDefinition cd = (LongRefillsCohortDefinition) cohortDefinition;
+
+        String startDate = DateUtil.formatDate(cd.getStartDate(), "yyyy-MM-dd");
+        String endDate = DateUtil.formatDate(cd.getEndDate(), "yyyy-MM-dd");
+
+        String q ="select e.patient_id from obs o inner  join  \n " +
```

Comment:
@balukujr Can you add a description to what this query does - not clear to me

Reply:
Done